eventauth.go

```diff
@@ -859,35 +857,37 @@ func (e *eventAllower) commonChecks(event PDU) error {
 		)
 	}
 
-	sender := event.Sender()
 	stateKey := event.StateKey()
-
-	if err := e.create.UserIDAllowed(sender); err != nil {
+	userID, err := event.UserID()
+	if err != nil {
+		return err
+	}
+	if err := e.create.UserIDAllowed(userID); err != nil {
```

Comment:
AKA "can we federate with this user ID?" which is unenforcable for pseudo IDs, so we'd remove it.